src/ngx_http_lua_logby.c

```diff
@@ -67,9 +70,39 @@ ngx_http_lua_log_by_lua_env(lua_State *L, ngx_http_request_t *r)
 ngx_int_t
 ngx_http_lua_log_handler(ngx_http_request_t *r)
 {
+#if NGX_HTTP_LUA_HAVE_MALLOC_TRIM
+    ngx_uint_t                   trim_cycle, trim_nreq;
+#endif
+    ngx_http_lua_main_conf_t    *lmcf;
     ngx_http_lua_loc_conf_t     *llcf;
     ngx_http_lua_ctx_t          *ctx;
 
+    lmcf = ngx_http_get_module_main_conf(r, ngx_http_lua_module);
+
+#if NGX_HTTP_LUA_HAVE_MALLOC_TRIM
+    trim_cycle = lmcf->malloc_trim_cycle;
+
+    if (trim_cycle > 0) {
+
+        dd("cycle: %d", (int) trim_cycle);
+
+        trim_nreq = ++lmcf->malloc_trim_req_count;
+
+        if (trim_nreq >= trim_cycle) {
+            lmcf->malloc_trim_req_count = 0;
+
+            ngx_log_debug1(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
+                           "malloc_trim(1) returned %d", malloc_trim(1));
+        }
```

Comment:
I could be wrong. if the '#if NGX_DEBUG' evaluates to false, none of the actual arguments to the ngx_log_debug1, meaning the malloc_trim() is not called at all. Other than this one, LGTM.